lib/dor/services/client/objects.rb

```diff
@@ -14,11 +14,10 @@ class Objects < VersionedService
         # @return [Cocina::Models::DROWithMetadata,Cocina::Models::CollectionWithMetadata,Cocina::Models::AdminPolicyWithMetadata] the returned model
         def register(params:, assign_doi: false, validate: false)
           resp = connection.post do |req|
-            req.url objects_path
+            req.url with_querystring(url: objects_path, params: { assign_doi: assign_doi })
             req.headers['Content-Type'] = 'application/json'
             # asking the service to return JSON (else it'll be plain text)
             req.headers['Accept'] = 'application/json'
-            req.params[:assign_doi] = true if assign_doi
             req.body = params.to_json
```

Comment:
similar question here about leaving `req.body = ...`

Reply:
No, these are different params (but unfortunately have the same argument name, so confusing).